src/WebJobs.Script/Extensions/ProjectExtensions.cs

```diff
@@ -1,52 +1,166 @@
+
+            group.AppendChild(project.CreatePackageReference(packageId, version));
+        }
+
+        public static void RemovePackageReference(this XmlDocument project, string packageId)
+        {
+            var projectElements = project.SelectNodes("//*").OfType<XmlElement>();
+
+            XmlElement existingPackageReference = projectElements
+                .FirstOrDefault(item => item?.Name == PackageReferenceElementName && item?.Attributes[PackageReferenceIncludeElementName]?.Value == packageId);
+
+            if (existingPackageReference != null)
+            {
+                existingPackageReference.ParentNode?.RemoveChild(existingPackageReference);
+            }
+        }
+
+        public static void AddTargetFramework(this XmlDocument project, string innerText)
+        {
+            var projectElements = project?.SelectNodes("//*").OfType<XmlElement>();
+
+            XmlElement existingPackageReference = projectElements
+                .FirstOrDefault(item => item?.Name == TargetFrameworkElementName && item.InnerText == innerText);
+
+            if (existingPackageReference != null)
+            {
+                return;
             }
 
-            ProjectItemGroupElement group = GetUniformItemGroupOrNew(project, PackageReferenceElementName);
+            XmlElement group = GetUniformItemGroupOrNew(project, PropertyGroupElementName);
 
-            group.AddItem(PackageReferenceElementName, packageId)
-                 .AddMetadata(PackageReferenceVersionElementName, version, true);
+            group.AppendChild(project.CreateTargetFramework(innerText));
         }
 
-        public static void RemovePackageReference(this ProjectRootElement project, string packageId)
+        public static void RemoveTargetFramework(this XmlDocument project, string innerText)
         {
-            ProjectItemElement existingPackageReference = project.Items
-                .FirstOrDefault(item => item.ItemType == PackageReferenceElementName && item.Include == packageId);
+            var projectElements = project.SelectNodes("//*").OfType<XmlElement>();
+
+            XmlElement existingPackageReference = projectElements
+                .FirstOrDefault(item => item?.Name == TargetFrameworkElementName && item.InnerText == innerText);
 
             if (existingPackageReference != null)
             {
-                existingPackageReference.Parent.RemoveChild(existingPackageReference);
+                existingPackageReference.ParentNode?.RemoveChild(existingPackageReference);
             }
         }
 
-        public static ProjectItemGroupElement GetUniformItemGroupOrNew(this ProjectRootElement project, string itemName)
+        private static XmlElement GetUniformItemGroupOrNew(this XmlDocument project, string itemName)
         {
-            ProjectItemGroupElement group = project.ItemGroupsReversed.FirstOrDefault(g => g.Items.All(i => i.ItemType == itemName));
```

Comment:
How important is it that we're using a reversed item group list here? I ask as we don't have an equivalent available through XmlDocument, so we would need to recreate our own reversed list of XmlElements if this is important. Probably through getting all elements and reversing it through `project.SelectNodes("//*").OfType<XmlElement>();`